e2e-tests/cypress/tests/integration/channels/notifications/reset_notification_when_settings_canceled_spec.js

```diff
@@ -0,0 +1,125 @@
+        openSettingsAndChangeNotification('callsDesktopSound');
+    });
+
+    function openSettingsAndChangeNotification(type) {
+        // # Open 'Settings' modal
+        cy.uiOpenSettingsModal().within(() => {
+            // # Navigate to Desktop Notification Settings
+            navigateToDesktopNotificationSettings(type);
+
+            // # Change Notification selection
+            setNotificationToDown(type);
+
+            // # Click Cancel button
+            cy.uiCancelButton().click();
+
+            // # Navigate to Desktop Notification Settings
+            navigateToDesktopNotificationSettings(type);
+        });
+    }
+
+    function setNotificationToDown(type) {
+        switch (type) {
+        case 'desktopNotification':
+            // # Change Notification sound selection value is set to Down
+            cy.get('#displaySoundNotification').
+                find('.react-select__dropdown-indicator').
+                click().
+                get('.react-select__menu').
+                contains('Down').
+                click();
+            break;
+        case 'callsDesktopSound':
+            // # Change Notification Notification sound for incoming calls selection value is set to Down
+            cy.get('#displayCallsSoundNotification').
+                find('.react-select__dropdown-indicator').
+                click().
+                get('.react-select__menu').
+                contains('Down').
+                click();
+            break;
+        default:
+            break;
+        }
+
+        // * Verify Notification display changed to Down
+        verifyNotificationSelectionValue(type, 'Down');
+    }
+
+    function navigateToDesktopNotificationSettings(type) {
+        // # Click on the 'Edit' button next to Desktop Notifications
+        cy.get('#desktopEdit').should('be.visible').click();
+
+        // * Verify that the Notification is set to On
+        verifyNotificationIsOn(type);
+
+        // * Verify Notification selection display default value (Bing)
+        verifyNotificationSelectionValue(type, 'Bing');
+    }
+
+    function verifyNotificationIsOn(type) {
+        switch (type) {
+        case 'desktopNotification':
+            // * Verify that the Notification sound is set to On
+            cy.get('#soundOn').should('be.visible').and('be.checked');
+            break;
+        case 'callsDesktopSound':
+            // * Verify that the Notification sound for incoming calls is set to On
+            cy.get('#callsSoundOn').should('be.visible').and('be.checked');
+            break;
+        default:
+            break;
+        }
+    }
+
+    function verifyNotificationSelectionValue(type, value) {
+        switch (type) {
+        case 'desktopNotification':
+            // * Verify that the Notification sound is set to certain value
+            cy.get('#displaySoundNotification').
+                find('.react-select__single-value').
```

Comment:
can we Use `cy.findByText(value).should(...)` please ? 
we usually avoid `.class` and chaining the commands to assert text.

Reply:
For react-select selection, I follow their sample spec test here: I just follow react-select e2e test sample here: https://github.com/JedWatson/react-select/blob/master/cypress/integration/single-select.spec.ts.

Reply:
Sure, I fix it in my new commit.